src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1076,15 +1076,22 @@ void InterpreterCreateQuery::assertOrSetUUID(ASTCreateQuery & create, const Data
 {
     const auto * kind = create.is_dictionary ? "Dictionary" : "Table";
     const auto * kind_upper = create.is_dictionary ? "DICTIONARY" : "TABLE";
+    bool is_replicated_database_internal = database->getEngineName() == "Replicated" && getContext()->getClientInfo().is_replicated_database_internal;
+    bool from_path = create.attach_from_path.has_value();
+    bool is_on_cluster = getContext()->getClientInfo().query_kind == ClientInfo::QueryKind::SECONDARY_QUERY;
 
-    if (database->getEngineName() == "Replicated" && getContext()->getClientInfo().is_replicated_database_internal
-        && !internal)
+    if (is_replicated_database_internal && !internal)
     {
         if (create.uuid == UUIDHelpers::Nil)
             throw Exception(ErrorCodes::LOGICAL_ERROR, "Table UUID is not specified in DDL log");
     }
 
-    bool from_path = create.attach_from_path.has_value();
+    if (create.refresh_strategy && database->getEngineName() != "Atomic" && !internal && !is_replicated_database_internal && !is_on_cluster)
```

Comment:
Considering that executing it with any other DB engine will cause a segfault I would simplify the condition:
```suggestion
    if (create.refresh_strategy && database->getEngineName() != "Atomic" && !internal)
```

Reply:
For example, after the update with support for Replicated databases we don't want a successfully submitted DDL query to segfault on an old replica. There is also a case where we can submit to DDL queue a query with the database that is not present on a leader replica, so although mostly not possible in real usage it may still cause a segfault.

Reply:
The condition indeed looks complicated, but it is the same as 
```
if (create.refresh_strategy && database->getEngineName() != "Atomic" && !(internal || is_replicated_database_internal || is_on_cluster)
```
But then I have a question about `is_on_cluster` - why do we need to check? AFAIK you cannot write a CREATE query with ON CLUSTER clause which creates a table inside Replicated database. 
P.S. Since it is a temporary fix - it is Ok to merge it.